clients/restore/restore.go

```diff
@@ -0,0 +1,292 @@
+package restore
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"path/filepath"
+	"sort"
+	"strings"
+
+	"github.com/influxdata/influx-cli/v2/api"
+	"github.com/influxdata/influx-cli/v2/clients"
+	br "github.com/influxdata/influx-cli/v2/internal/backup_restore"
+	"github.com/influxdata/influx-cli/v2/pkg/gzip"
+)
+
+type Client struct {
+	clients.CLI
+	api.RestoreApi
+	api.OrganizationsApi
+
+	manifest br.Manifest
+}
+
+type Params struct {
+	// Path to local backup data created using `influx backup`
+	Path string
+
+	// Original ID/name of the organization to restore.
+	// If not set, all orgs will be restored.
+	OrgID string
+	Org   string
+
+	// New name to use for the restored organization.
+	// If not set, the org will be restored using its backed-up name.
+	NewOrgName string
+
+	// Original ID/name of the bucket to restore.
+	// If not set, all buckets within the org filter will be restored.
+	BucketID string
+	Bucket   string
+
+	// New name to use for the restored bucket.
+	// If not set, the bucket will be restored using its backed-up name.
+	NewBucketName string
+
+	// If true, replace all data on the server with the local backup.
+	// Otherwise only restore the requested org/bucket, leaving other data untouched.
+	Full bool
+}
+
+func (p *Params) matches(bkt br.ManifestBucketEntry) bool {
+	if p.OrgID != "" && bkt.OrganizationID != p.OrgID {
+		return false
+	}
+	if p.Org != "" && bkt.OrganizationName != p.Org {
+		return false
+	}
+	if p.BucketID != "" && bkt.BucketID != p.BucketID {
+		return false
+	}
+	if p.Bucket != "" && bkt.BucketName != p.Bucket {
+		return false
+	}
+	return true
+}
+
+func (c *Client) Restore(ctx context.Context, params *Params) error {
+	if err := c.loadManifests(params.Path); err != nil {
+		return err
+	}
+	if params.Full {
+		return c.fullRestore(ctx, params.Path)
+	}
+	return c.partialRestore(ctx, params)
+}
+
+// loadManifests finds and merges all backup manifests stored in a given directory,
+// keeping the latest top-level metadata and latest metadata per-bucket.
+func (c *Client) loadManifests(path string) error {
+	// Read all manifest files from path, sort in ascending time.
+	manifests, err := filepath.Glob(filepath.Join(path, fmt.Sprintf("*.%s", br.ManifestExtension)))
+	if err != nil {
+		return fmt.Errorf("failed to find backup manifests at %q: %w", path, err)
+	} else if len(manifests) == 0 {
+		return fmt.Errorf("no backup manifests found at %q", path)
+	}
+	sort.Strings(manifests)
+
+	bucketManifests := map[string]br.ManifestBucketEntry{}
+	for _, manifestFile := range manifests {
+		// Skip file if it is a directory.
```

Comment:
Is it possible for this to happen? I guess if there was a directory called `whatever.manifest` ?

Reply:
Yeah, I wouldn't expect it to happen but it's in the 2.0.x implementation so I'm guessing somebody got bitten by a weird filesystem layout in the past.